google-cloud-firestore/lib/google/cloud/firestore/document/snapshot.rb

```diff
@@ -0,0 +1,341 @@
+          #
+          #   nyc_snap = firestore.get "cities/NYC"
+          #
+          #   nyc_snap.cols.each do |col|
+          #     # Print the collection
+          #     puts col.collection_id
+          #   end
+          #
+          def cols &block
+            ref.cols(&block)
+          end
+          alias_method :collections, :cols
+
+          ##
+          # Retrieves a collection nested under the document with data.
+          #
+          # @param [String] collection_path A string representing the path of
+          #   the collection, relative to the document.
+          #
+          # @return [Collection::Reference] A collection.
+          #
+          # @example
+          #   require "google/cloud/firestore"
+          #
+          #   firestore = Google::Cloud::Firestore.new
+          #
+          #   nyc_snap = firestore.get "cities/NYC"
+          #
+          #   # Get precincts sub-collection
+          #   precincts_col = nyc_snap.col "precincts"
+          #
+          def col collection_path
+            ref.col collection_path
+          end
+          alias_method :collection, :col
+
+          ##
+          # Retrieves the document data.
+          #
+          # @return [Hash] The document data.
+          #
+          # @example
+          #   require "google/cloud/firestore"
+          #
+          #   firestore = Google::Cloud::Firestore.new
+          #
+          #   nyc_snap = firestore.get "cities/NYC"
+          #
+          #   # Get the document data
+          #   nyc_snap.data[:population] #=> 1000000
+          #
+          def data
+            return nil if missing?
+            Convert.fields_to_hash grpc.fields, ref.context
+          end
+          alias_method :fields, :data
+
+          ##
+          # Retrieves the document data.
+          #
+          # @param [String|Symbol, Array<String|Symbol>] field_path A string,
+          #   symbol, or array representing the path of the data to select.
+          #
+          # @return [Object] The data at the field path.
+          #
+          # @example
+          #   require "google/cloud/firestore"
+          #
+          #   firestore = Google::Cloud::Firestore.new
+          #
+          #   nyc_snap = firestore.get "cities/NYC"
+          #
+          #   # Get the document data
+          #   nyc_snap.get(:population) #=> 1000000
+          #
+          # @example Accessing data using []:
+          #   require "google/cloud/firestore"
+          #
+          #   firestore = Google::Cloud::Firestore.new
+          #
+          #   nyc_snap = firestore.get "cities/NYC"
+          #
+          #   # Get the document data
+          #   nyc_snap[:population] #=> 1000000
+          #
+          # @example Nested data can be accessing with field path:
+          #   require "google/cloud/firestore"
+          #
+          #   firestore = Google::Cloud::Firestore.new
+          #
+          #   frank_snap = firestore.get "users/frank"
```

Comment:
😸

Reply:
I wish we could take credit for that code example, but that comes from [here](https://cloud.google.com/firestore/docs/manage-data/add-data#update-data).

Reply:
I'm sure there's a 12 year old out there who feels better about this information being in a sample 😄 

~~ end ~~